workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/ProductCard.tsx

```diff
@@ -3,131 +3,216 @@
 
 interface ProductCardProps {
 	productId: number | null;
 	selectedAccount?: Account;
 	setProductToForm: (product: Product) => void;
-	showAccount: Boolean;
 }
 
 const ProductCard = ({
 	productId,
 	selectedAccount,
 	setProductToForm,
-	showAccount,
 }: ProductCardProps) => {
-	const [product, setProduct] = useState<Product[]>([]);
+	const [product, setProduct] = useState<Product>();
+	const [hasTrial, setHasTrial] = useState<boolean>(false);
+	const [basePrice, setBasePrice] = useState<Number | undefined>(undefined);
+
+	const productHasTrialSKU = (skus: SKU[]) => {
+		skus.forEach((sku) => {
+			const licenseUsageType = sku.skuOptions.find(
+				(option) =>
+					option.key.toLowerCase() === 'dxp-license-usage-type'
+			);
+			if (
+				licenseUsageType &&
+				licenseUsageType.value.toLowerCase() ===
+					SkuOptions.TRIAL.toLowerCase()
+			) {
+				setHasTrial(true);
+			}
+		});
+	};
+
+	const getProductBasePrice = (product: Product) => {
+		product &&
+			product.skus.forEach((sku) => {
+				const licenseUsageType = sku.skuOptions.find(
+					(option) =>
+						option.key.toLowerCase() === 'dxp-license-usage-type'
+				);
+				if (
+					licenseUsageType &&
+					licenseUsageType.value.toLowerCase() ===
+						SkuOptions.STANDARD.toLowerCase()
+				) {
+					setBasePrice(sku.price);
+				}
+			});
+	};
 
 	useEffect(() => {
-		const getProdut = async () => {
-			// eslint-disable-next-line promise/catch-or-return
-			{
+		const fetchData = async () => {
+			const productResponse =
 				productId &&
-					getProductById({
-						nestedFields: 'skus',
-						productId,
-					}).then((item: Product) => {
-						setProduct([item]);
-					});
+				(await getProductById({
+					nestedFields: 'attachments,productSpecifications,skus',
+					productId,
+				}));
+
+			if (productResponse) {
+				setProduct(productResponse);
+				productHasTrialSKU(productResponse.skus);
+				setProductToForm(productResponse);
+				getProductBasePrice(productResponse);
 			}
 		};
-		getProdut();
-	}, [productId]);
 
-	setProductToForm(product[0]);
+		fetchData();
+	}, [productId, setProductToForm]);
 
-	const currentValue = product[0];
+	const iconURL =
+		product &&
+		getThumbnailByProductAttachment(product.attachments)?.split('/o/');
+	const convertedIconURL = iconURL ? `/o/${iconURL[1]}` : '';
 
-	let setHeight;
+	const getLicenseTagText = (product: Product) => {
+		if (
+			getValueFromSpecifications(
+				product.productSpecifications,
+				'license-type'
+			).toLowerCase() === LicenseType.Perpetual
+		) {
+			return 'One-Time';
+		}
+		else if (
+			getValueFromSpecifications(
+				product.productSpecifications,
+				'license-type'
+			).toLowerCase() === LicenseType.Subscription
+		) {
+			return 'Annually';
+		}
 
-	if (showAccount) {
-		setHeight = 176;
-	}
-	else {
-		setHeight = 112;
-	}
+		return '';
+	};
+
+	const getPriceText = (product: Product) => {
+		if (
+			getValueFromSpecifications(
+				product.productSpecifications,
+				'price'
```

Comment:
Shouldn't we change it back ([PR](https://github.com/is-solutions-delivery/liferay-portal/pull/4841/files)) to
```suggestion
				'price-model'
```
?